libnavui-voice/src/main/java/com/mapbox/navigation/ui/views/soundbutton/SoundButton.kt

```diff
@@ -0,0 +1,259 @@
+package com.mapbox.navigation.ui.views.soundbutton
+
+import android.content.Context
+import android.content.res.ColorStateList
+import android.content.res.TypedArray
+import android.util.AttributeSet
+import android.view.animation.AccelerateInterpolator
+import android.view.animation.AlphaAnimation
+import android.view.animation.Animation
+import android.view.animation.AnimationSet
+import android.view.animation.DecelerateInterpolator
+import android.widget.TextView
+import androidx.annotation.StyleRes
+import androidx.constraintlayout.widget.ConstraintLayout
+import androidx.core.content.ContextCompat
+import androidx.core.graphics.drawable.DrawableCompat
+import com.google.android.material.floatingactionbutton.FloatingActionButton
+import com.mapbox.navigation.ui.base.internal.utils.ClickListenersDelegate
+import com.mapbox.navigation.ui.voice.R
+
+/**
+ * Sound Button view.
+ */
+class SoundButton : ConstraintLayout {
+
+    private lateinit var soundFab: FloatingActionButton
+    private lateinit var soundChipText: TextView
+    private var fadeInSlowOut: AnimationSet? = null
+    private var clickListenersDelegate: ClickListenersDelegate? = ClickListenersDelegate()
+
+    private var primaryColor = 0
+    private var secondaryColor = 0
+
+    /**
+     * *true* if it's muted, otherwise *false*
+     */
+    var isMuted = false
+        private set
+
+    private companion object {
+        private const val ALPHA_VALUE_ZERO = 0f
+        private const val ALPHA_VALUE_ONE = 1f
+        private const val ANIMATION_DURATION_THREE_HUNDRED_MILLIS: Long = 300
+        private const val ANIMATION_DURATION_ONE_THOUSAND_MILLIS: Long = 1000
+    }
+
+    constructor(context: Context) : super(context) {
+        init(context)
+    }
+
+    constructor(context: Context, attrs: AttributeSet?) : super(context, attrs) {
```

Comment:
Please take a look on how `MapboxTripProgressView` and `MapboxManeuverView` has been implemented. We can remove all the constructors and instead just have one. Also it would be great to maintain consistency for `initAttributes`, `applyAttributes` and `updateStyle` functions.

Reply:
please have a look https://github.com/mapbox/mapbox-navigation-android/issues/4134

Reply:
The ticket discusses this particular point > `Please take a look on how MapboxTripProgressView and MapboxManeuverView has been implemented. We can remove all the constructors and instead just have one.`

But we should still make sure to use `initAttributes`, `applyAttributes` and `updateStyle` for consistency sake.